dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h

```diff
@@ -743,13 +750,14 @@ class nvJPEGDecoder : public Operator<MixedBackend>, CachedDecoderImpl {
       in_data_.resize(samples_hw_batched_.size());
       in_lengths_.resize(samples_hw_batched_.size());
       nvjpeg_destinations_.resize(samples_hw_batched_.size());
+      nvjpeg_params_.resize(samples_hw_batched_.size());
       memset(nvjpeg_destinations_.data(), 0, nvjpeg_destinations_.size() * sizeof(nvjpegImage_t));
+      memset(nvjpeg_params_.data(), 0, nvjpeg_params_.size() * sizeof(nvjpegDecodeParams_t));
```

Comment:
```suggestion
      nvjpeg_params_.clear();
      nvjpeg_params_.resize(samples_hw_batched_.size());
      memset(nvjpeg_destinations_.data(), 0, nvjpeg_destinations_.size() * sizeof(nvjpegImage_t));
```
This will fill the vector with default-initialized objects, potentially avoid a copy (if reallocating) and is immune to replacing the element type with a non-POD.